### PR TITLE
Improve /tadmin delete behavior

### PR DESCRIPTION
**Pull request by Technoguyfication**
It looks like occasionally players don't get teleported out of a TARDIS in deletion quickly enough to not get cleared with the rest of the entities inside. Trying to remove player entities causes an error and the deletion ends up failing.

This PR adds a few things:
- Adds an option to teleport players out of the door instantaneously rather than scheduling for 10 ticks later
- Fixes the double teleport bug every time you enter or exit the TARDIS, making walking in and out a lot smoother
- Allows admins to delete TARDISes while inside them, since it is perfectly safe to be inside a TARDIS while its being deleted now.

Seeing your extermination rewrite, it might be useful to apply some of these changes to extermination as well. It might be a good idea to force players to stand outside their TARDIS while exterminating it anyways, but we can at least ensure that any companions or other travelers get safely teleported out.

Note: I also touched a completed unrelated section of code. VSCode was complaining about redeclaring a variable in a lambda function, so I simply changed the name of it.

### CODE REVIEW

File: src/main/java/me/eccentric_nz/TARDIS/builders/TARDISInteriorPostioning.java [hunk 1]
```diff
@@ -232,7 +232,7 @@ public void reclaimChunks(World w, int id) {
                             }
                             // remove dropped items
                             for (Entity e : tipsChunk.getEntities()) {
-                                e.remove();
+                                //e.remove();
```
Comment (eccentricdevotion) [inline src/main/java/me/eccentric_nz/TARDIS/builders/TARDISInteriorPostioning.java:235]: Are we not wanting to remove item drops that occur when blocks are removed eg torches dropping when the block they are attached is is removed? They will eventually despawn, but will cause some lag while they're still around? or do we not need to bother as they're removed when they fall into the void, I can never remember...
If we don't need to remove item entities then there's no point looping through them and all three lines can be removed.

**Comment by eccentricdevotion** [comment]
Players can't delete their own TARDISes unless they are [empty of other players ](https://github.com/eccentricdevotion/TARDIS/blob/v4.0/src/main/java/me/eccentric_nz/TARDIS/destroyers/TARDISExterminator.java#L145)

File: src/main/java/me/eccentric_nz/TARDIS/builders/TARDISInteriorPostioning.java (hunk 1 above)
Comment (Technoguyfication) [reply src/main/java/me/eccentric_nz/TARDIS/builders/TARDISInteriorPostioning.java:235]: Ah, I think I forgot to uncomment that line, actually. Let me fix that and test real quick!

File: src/main/java/me/eccentric_nz/TARDIS/commands/admin/TARDISDeleteCommand.java [hunk 2]
```diff
@@ -75,19 +75,6 @@ boolean deleteTARDIS(CommandSender sender, String[] args) {
...
-                if (player.getName().equals(args[1])) {
-                    HashMap<String, Object> wherep = new HashMap<>();
-                    wherep.put("uuid", player.getUniqueId().toString());
-                    ResultSetTravellers rst = new ResultSetTravellers(plugin, wherep, false);
-                    if (rst.resultSet()) {
-                        TARDISMessage.send(sender, "TARDIS_DELETE_NO");
-                        return true;
-                    }
-                }
-            }
```
Comment (eccentricdevotion) [inline src/main/java/me/eccentric_nz/TARDIS/commands/admin/TARDISDeleteCommand.java:90]: Technically I could now delete my own TARDIS while inside it...

Comment (Technoguyfication) [reply src/main/java/me/eccentric_nz/TARDIS/commands/admin/TARDISDeleteCommand.java:90]: Correct. It seems to work perfectly fine from my testing, you get teleported out the instant the interior deletes and the police box starts dematerializing. Here's a short clip of how it looks from the inside and outside: https://tkno.pw/3q0fju0

**Comment by Technoguyfication** [comment]
Also something I realized in your first review about deleting dropped items, depending on how aggressively the server unloads chunks the items may not have time to fall into the void before the chunks are unloaded and saved. This would cause the items to appear in the next TARDIS built in that slot. It's happened to me on older versions of the plugin, at least.

**Comment by Technoguyfication** [comment]
Actually might want to wait a bit before merging this PR. It looks like the barrier blocks under the door aren't replaced back to their original blocks when the TARDIS is deleted. I'm going to investigate and see if I can fix that in this pull request.

File: src/main/java/me/eccentric_nz/TARDIS/commands/admin/TARDISDeleteCommand.java (hunk 2 above)
Comment (eccentricdevotion) [reply src/main/java/me/eccentric_nz/TARDIS/commands/admin/TARDISDeleteCommand.java:90]: Okay cool

**Comment by Technoguyfication** [comment]
Okay, I can't figure out why it's not replacing the under door blocks when removing a TARDIS. It does it fine with travelling and hiding, but not when deleting or exterminating. Any ideas?

**Comment by Technoguyfication** [comment]
> Add to [TARDISDestroyerInner.java](https://github.com/eccentricdevotion/TARDIS/blob/v4.0/src/main/java/me/eccentric_nz/TARDIS/destroyers/TARDISDestroyerInner.java#L72)
> 
> ```
> where.put("tardis_id", id);
> where.put("police_box", 0);  <----------------------------------- add this line
> plugin.getQueryFactory().doDelete("blocks", where);
> ```
> 
> Should fix it

Hmm, I've added that line and recompiled the plugin but it's still leaving the barrier blocks behind when the TARDIS is deleted. I tried growing a new one after making the change too, same effect unfortunately.

I might be wrong, but it looks like that line of code is just removing the entries from the `blocks` table in the database, rather than actually replacing the blocks with the original ones.

**Comment by eccentricdevotion** [comment]
Hmm, that's interesting... there must be somewhere else that is deleting the database records that hold the data to restore the blocks. I did fix it in my testing with another method, but it didn't really address the root of the problem  and we do without more ugly hacks in the TARDIS code lol
Maybe also remove `"blocks", ` from the `cleanDatabase(int id)` method in [TARDISDeleteCommand.java](https://github.com/eccentricdevotion/TARDIS/blob/v4.0/src/main/java/me/eccentric_nz/TARDIS/commands/admin/TARDISDeleteCommand.java#L57)

**Comment by Technoguyfication** [comment]
Well, I took the `"blocks", ` out from the `cleanDatabase` method in both the `TARDISDeleteCommand` and `TARDISExterminator` and it's still not getting rid of the barriers when doing either. Definitely looks like they're getting deleted somewhere still

**Comment by eccentricdevotion** [comment]
Ok, in my testing it's working for '/tardis exterminate` but not for `/tadmin delete` which narrows it down a bit...

**Comment by eccentricdevotion** [comment]
Working as expected now, barrier blocks are removed.
[Block restoration](https://www.youtube.com/watch?v=FZZ8-nyex8Y "Block restoration")

**Comment by Technoguyfication** [comment]
Awesome! Should I pull those changes into my fork?

**Comment by eccentricdevotion** [comment]
I'll push them now...

**Comment by Technoguyfication** [comment]
Rebased my branch off of yours, should be up to date now!

**Comment by eccentricdevotion** [comment]
Outstanding, thanks again :)

**Pull request merged**